### PR TITLE
[DO NOT SQUASH] SCSCANGHA-25 Remove docker and SCSCANGHA-28 Rebranding

### PR DESCRIPTION
**Pull request by antonioaversa**
This PR contains two tasks:
- SCSCANGHA-25 Replace the Docker action by a composite action forwarding to the SonarQube action
  - implemented as forward to `sonarqube-scan-action@v4.1.0` **(fully-fixed version)**
- SCSCANGHA-28 Rebranding
  - Main changes are in the [README.md](https://github.com/SonarSource/sonarcloud-github-action/blob/antonio/SCSCANGHA-25-28-remove-docker-and-rebranding/README.md) and in the [action.yml](https://github.com/SonarSource/sonarcloud-github-action/blob/antonio/SCSCANGHA-25-28-remove-docker-and-rebranding/action.yml) 
 
We have also added a warning:
- in the README
- in the logs
- in the annotations

image
image
image

Successful test: https://github.com/antonioaversa/dart-tools-test2/actions/runs/12086908914/job/33707231926?pr=1

### CODE REVIEW

File: .github/workflows/qa.yml
Comment (antonioaversa) [inline .github/workflows/qa.yml:1]: Given that this action is going to be deprecated, and all it does is to forward to `sonarqube-scan-action`, I have only kept the tests about the input parameters of the action, which ensure that all the input parameters are forwarded to the `sonarqube-scan-action` accordingly.
All the other tests are done in the `sonarqube-scan-action`, including the one on SC Url env var.

File: README.md [hunk 1]
```diff
@@ -1,127 +1,151 @@
...
+> The `sonarqube-scan-action` is a drop-in replacement for this action.
 
-<img src="./images/SonarCloud-72px.png">
+This SonarSource project, available as a GitHub Action, scans your projects with SonarQube [Cloud](https://www.sonarsource.com/products/sonarcloud/).
 
-[SonarCloud](https://www.sonarsource.com/products/sonarcloud/) is a widely used static analysis solution for continuous code quality and security inspection. 
-It helps developers identify and fix issues in their code that could lead to bugs, vulnerabilities, or decreased development velocity.
-SonarCloud supports the most popular programming languages, including Java, JavaScript, TypeScript, C#, Python, C, C++, and [many more](https://www.sonarsource.com/knowledge/languages/).
+![Logo](./images/SQ_Logo_Cloud_Dark_Backgrounds.png#gh-dark-mode-only)
+![Logo](./images/SQ_Logo_Cloud_Light_Backgrounds.png#gh-light-mode-only)
```
Comment (antonioaversa) [inline README.md:11]: Having both dart and light logos with `#gh-dark-mode-only` and `#gh-light-mode-only` fragments make the logo responsive to the theme of the user.

File: test/gradle-project-kotlin/build.gradle.kts
Comment (antonioaversa) [inline test/gradle-project-kotlin/build.gradle.kts:1]: All these test files were used by QA workflow tests which now are only executed in the QA of `sonarqube-scan-action`. So I have removed them.

File: README.md
Comment (antonioaversa) [inline README.md:1]: This is in line with the [README of `sonarqube-scan-action`](https://github.com/SonarSource/sonarqube-scan-action/blob/dev/README.md), except that there are only references to SonarQube Cloud.

File: action.yml [hunk 2]
```diff
@@ -1,20 +1,38 @@
...
+    default: https://binaries.sonarsource.com/Distribution/sonar-scanner-cli
+runs:
+  using: "composite"
+  steps:
+    - name: Deprecation warning
+      shell: bash
+      run: |
+        echo "::warning title=SonarScanner::This action is deprecated and will be removed in a future release. Please use the sonarqube-scan-action action instead. The sonarqube-scan-action is a drop-in replacement for this action."
+    - name: SonarQube Cloud Scan
+      uses: SonarSource/sonarqube-scan-action@dev # TODO: fix me to v4.1.0
```
Comment (antonioaversa) [inline action.yml:33]: This has to be changed to point to `v4.1.0`, once `sonarqube-scan-action@v4.1.0` is released.

**Comment by antonioaversa** [comment]
Adding @agigleux to validate the warning message (check description in the PR).

File: action.yml (hunk 2 above)
Comment (antonioaversa) [reply action.yml]: Done

**Review by henryju — APPROVED** [review]
LGTM!

**Pull request merged**